unittests/rpc/BaseTests.cpp

```diff
@@ -220,6 +220,44 @@ TEST_F(RPCBaseTest, ArrayAtValidator)
     ASSERT_FALSE(spec.validate(failingInput));
 }
 
+TEST_F(RPCBaseTest, IfTypeValidator)
+{
+    // clang-format off
+    auto spec = RpcSpec{
+        {"mix", Required{}, 
+                Type<std::string,json::object>{},
+                IfType<json::object>{
+                        Section{{ "limit", Required{}, Type<uint32_t>{}, Between<uint32_t>{0, 100}}, },
+                        Section{{ "limit2", Required{}, Type<uint32_t>{}, Between<uint32_t>{0, 100}}, },
+                        },
```

Comment:
minor: Formatting looks a bit messed up and i'd remove the extra commas at the end of the list where they are not needed.